### PR TITLE
[screloc] small improvements

### PR DESCRIPTION
**Pull request by luca-s**
Please see each individual commits for details:

[screloc] minor code logic fix
[screloc] --ep honours ignoreRejected option as well
[screloc] add option --dropfailure (for --ep processing)
[screloc] log how many origins were successfully relocated

### CODE REVIEW

**Comment by luca-s** [comment]
@gempa-jabe ping...in case you missed this

File: apps/screloc/main.cpp [hunk 1]
```diff
@@ -81,6 +81,8 @@ class Reloc : public Client::Application {
 			                                       "This option should not be mixed with --dump.", &_epFile);
 			commandline().addOption("Input", "replace", "Used in combination with --ep and defines if origins are to be replaced "
 			                                            "by their relocated counterparts or just added to the output.");
+			commandline().addOption("Input", "dropfailure", "Used in combination with --replace/--ep and drops from the output "
```
Comment (gempa-jabe) [inline apps/screloc/main.cpp]: As we are using a dash to separate two words, I prefer `--drop-failure`.

File: apps/screloc/main.cpp [hunk 2]
```diff
@@ -270,35 +273,56 @@ class Reloc : public Client::Application {
...
 					}
 
+					bool relocated = false;
 					try {
 						org = process(org.get());
+						relocated = true;
 					}
 					catch ( std::exception &e ) {
 						SEISCOMP_ERROR("  + processing failed - %s", e.what());
+						relocated = false;
```
Comment (gempa-jabe) [inline apps/screloc/main.cpp:283]: No necessary as `false` is default.

File: apps/screloc/main.cpp [hunk 3]
```diff
@@ -270,35 +273,56 @@ class Reloc : public Client::Application {
...
+					}
+					if ( !org ) { // safety belt, but it should not happen
```
Comment (gempa-jabe) [inline apps/screloc/main.cpp:285]: As you do continue with `org == nullptr`, this case must be handled in all other lines where `org` is being accessed.

File: apps/screloc/main.cpp [hunk 4]
```diff
@@ -270,35 +273,56 @@ class Reloc : public Client::Application {
...
+						SEISCOMP_ERROR("  + processing failed %s", publicID.c_str());
+						relocated = false;
+					}
+					if ( _ignoreRejected ) {
+						try {
+							if ( org->evaluationStatus() == REJECTED ) {
```
Comment (gempa-jabe) [inline apps/screloc/main.cpp:291]: `org` can be `nullptr`.

File: apps/screloc/main.cpp [hunk 5]
```diff
@@ -270,35 +273,56 @@ class Reloc : public Client::Application {
...
-					if ( org ) {
-						if ( replace ) {
-							ep->removeOrigin(i);
-							--i;
-							--numberOfOrigins;
-						}
-
-						ep->add(org.get());
-					}
+					ep->add(org.get());
```
Comment (gempa-jabe) [inline apps/screloc/main.cpp:317]: `org` can be `nullptr`.

File: apps/screloc/main.cpp (hunk 1 above)
Comment (luca-s) [reply apps/screloc/main.cpp]: sounds good

File: apps/screloc/main.cpp (hunk 2 above)
Comment (luca-s) [reply apps/screloc/main.cpp:283]: I know, but I like pedantic/obvious code.  The compiler will remove it anyway.

File: apps/screloc/main.cpp (hunk 3 above)
Comment (luca-s) [reply apps/screloc/main.cpp:285]: I totally missed that, thanks.

File: apps/screloc/main.cpp (hunk 4 above)
Comment (luca-s) [reply apps/screloc/main.cpp:291]: This is the only case I need to handle with org being `nullptr`

File: apps/screloc/main.cpp (hunk 5 above)
Comment (luca-s) [reply apps/screloc/main.cpp:317]: a `nullptr` org never reaches here

Comment (gempa-jabe) [reply apps/screloc/main.cpp:317]: Correct, my bad.

**Comment by luca-s** [comment]
Everything should be fixed now

**Comment by luca-s** [comment]
Could we merge this too?

**Comment by luca-s** [comment]
Thanks

**Comment by luca-s** [comment]
I will merge this PR to master branch as well. Unfortunately I did the same mistake again of providing a PR based on branch v6

**Pull request merged**